PSFramework/internal/loggingProviders/gelf.provider.ps1

```diff
@@ -0,0 +1,159 @@
+$configurationParameters = {
+	$configroot = "psframework.logging.gelf"
+
+	$configurations = Get-PSFConfig -FullName "$configroot.*"
+
+	$RuntimeParamDic = New-Object  System.Management.Automation.RuntimeDefinedParameterDictionary
+
+	foreach ($config in $configurations)
+	{
+		$ParamAttrib = New-Object System.Management.Automation.ParameterAttribute
+		$ParamAttrib.ParameterSetName = '__AllParameterSets'
+		$AttribColl = New-Object System.Collections.ObjectModel.Collection[System.Attribute]
+		$AttribColl.Add($ParamAttrib)
+		$RuntimeParam = New-Object System.Management.Automation.RuntimeDefinedParameter(($config.FullName.Replace($configroot, "").Trim(".")), $config.Value.GetType(), $AttribColl)
+
+		$RuntimeParamDic.Add(($config.FullName.Replace($configroot, "").Trim(".")), $RuntimeParam)
+	}
+	return $RuntimeParamDic
+}
+
+# Script that is executes when configuring the provider using Set-PSFLoggingProvider
+$configurationScript = {
+	$configroot = "psframework.logging.gelf"
+
+	$configurations = Get-PSFConfig -FullName "$configroot.*"
+
+	foreach ($config in $configurations)
+	{
+		if ($PSBoundParameters.ContainsKey(($config.FullName.Replace($configroot, "").Trim("."))))
+		{
+			Set-PSFConfig -Module $config.Module -Name $config.Name -Value $PSBoundParameters[($config.FullName.Replace($configroot, "").Trim("."))]
+		}
+	}
+}
+
+# Script that returns a boolean value. "True" if all prerequisites are installed, "False" if installation is required
+$isInstalledScript = {
+	if (Get-Module -Name PSGELF -ListAvailable) {
+        return $true
+    }
+    else {
+        return $false
+    }
+}
+
+# Script that provides dynamic parameter for Install-PSFLoggingProvider
+$installationParameters = {
+	# None needed
+}
+
+# Script that performs the actual installation, based on the parameters (if any) specified in the $installationParameters script
+$installationScript = {
+    # install PSGELF module
+    if (-not (Get-Module -Name PSGELF -ListAvailable)) {
+        Install-Module -Name PSGELF
+    }
+}
+#endregion Function Extension / Integration
+
+# Configuration settings to initialize
+$configuration_Settings = {
+	Set-PSFConfig -Module PSFramework -Name 'Logging.GELF.GelfServer' -Value "" -Initialize -Validation string -Handler { } -Description "The path to where the logfile is written. Supports some placeholders such as %Date% to allow for timestamp in the name. For full documentation on the supported wildcards, see the documentation on https://psframework.org"
+	Set-PSFConfig -Module PSFramework -Name 'Logging.GELF.Port' -Value "" -Initialize -Validation string -Handler { } -Description "A special string you can use as a placeholder in the logfile path (by using '%logname%' as placeholder)"
+	Set-PSFConfig -Module PSFramework -Name 'Logging.GELF.Encrypt' -Value $true -Initialize -Validation bool -Handler { } -Description "Whether a written csv file will include headers"
```

Comment:
Descriptions of the settings still stem from the original settings they were copied from